### PR TITLE
client.invoke Sane Defaults (Decode & Input)

### PR DESCRIPTION
**Pull request by dOrgJelli**
Context: https://discord.com/channels/796821176743362611/814588073581084723/920812601036910704

### CODE REVIEW

**Review by krisbitney — DISMISSED** [review]
Everything looks good. I wonder if it would be better to change `decode` to something like `noDecode` and reverse the true/false usages, so that undefined is the same as false?

File: packages/js/core/src/__tests__/resolve-uri.spec.ts [hunk 1]
```diff
@@ -29,13 +29,9 @@ import {
 
 describe("resolveUri", () => {
   const client = (apis: Record<string, PluginModules>): Client => ({
-    getInvokeContext: (id: string) => {
```
Comment (nerfZael) [inline packages/js/core/src/__tests__/resolve-uri.spec.ts:32]: How did this build before? Aren't the "getRedirects" and other methods required on the Client?

File: packages/js/client/src/__tests__/Web3ApiClient.spec.ts [hunk 2]
```diff
@@ -121,6 +122,56 @@ describe("Web3ApiClient", () => {
...
+      const result = await client.invoke({
+        uri: ensUri,
+        module: "mutation",
+        method: "deployContract",
+        input: {
+          connection: {
+            networkNameOrChainId: "testnet"
+          }
+        },
+        decode: false
```
Comment (nerfZael) [inline packages/js/client/src/__tests__/Web3ApiClient.spec.ts]: It's usually not a great practice to have to set params to false to override behavior
Defaults should be false by default

**Review by nerfZael — CHANGES_REQUESTED** [review]

**Comment by nerfZael** [comment]
> Everything looks good. I wonder if it would be better to change `decode` to something like `noDecode` and reverse the true/false usages, so that undefined is the same as false?

I'm 100% on board with this, not sure about the naming though (but am not against it)

(Why can't I reply to this comment but have to quote it, GitHub is weird)

File: packages/js/core/src/__tests__/resolve-uri.spec.ts (hunk 1 above)
Comment (dOrgJelli) [reply packages/js/core/src/__tests__/resolve-uri.spec.ts:32]: It didn't build before :P Jest was throwing warnings, but still running the tests just fine.

File: packages/js/client/src/__tests__/Web3ApiClient.spec.ts (hunk 2 above)
Comment (dOrgJelli) [reply packages/js/client/src/__tests__/Web3ApiClient.spec.ts]: Great point, as discussed I'll change this to `noDecode` instead.

**Pull request merged**